### DISCUSSION TITLE
Payment Processing - What's Everyone Using? I need to switch ASAP!

### DISCUSSION THREAD

**Discussion by johnvoncolln** [OpenMage/magento-lts#3466] [General]
Hi,

I'm based in the US and currently using Paypal Payments Pro, but they have removed any fraud filters or AVS filters, which is proving to be a pain point. I've attempted several times to get Stripe working, but it's super buggy and inconsistent. This goes for the extension Stripe made and the one that Amasty created.

This has been discussed before, but it seems pretty darn important since you can't have a functional e-commerce site without a way for your customers to pay.

I'm eager to hear what anyone else is using. Please also post links to extensions and platforms!

**Comment by fballiano** [comment]
top of my head: hipay, braintree, SagePaymentsPro for 3 of my customers

**Reply by johnvoncolln** [reply]
@fballiano The braintree extension works well?

**Reply by johnvoncolln** [reply]
Also, do you have a link to the Braintree extension that you're using?

**Reply by fballiano** [reply]
a few links here: https://github.com/fballiano/awesome-openmage

this is the braintree ext by @justinbeaty https://github.com/justinbeaty/module-gene-braintree

**Reply by johnvoncolln** [reply]
awesome, much appreciated! Fingers crossed this is my solution

**Reply by justinbeaty** [reply]
@johnvoncolln it works great for me, as long as you manage your OpenMage install with composer.

**Reply by johnvoncolln** [reply]
@justinbeaty I've got it installed and working locally. Is there any information on migrating to using composer for openmage core updating instead of the old copy and merge workflow?

**Reply by justinbeaty** [reply]
@johnvoncolln This article may help: https://alanastorm.com/installing-openmage-bridging-a-generational-gap/

I use the "Aydin Hassan Installer" method, something as simple as putting the below into a `composer.json` file in an empty directory and then running `composer install` should work:

```json
{
  "minimum-stability": "dev",
  "prefer-stable": true,
  "repositories": [
    { "type": "vcs", "no-api": true, "url": "git@github.com:justinbeaty/module-gene-braintree.git" },
  ],
  "require": {
    "aydin-hassan/magento-core-composer-installer": "*",
    "openmage/magento-lts": "*",
    "gene/braintree": "dev-master",
  }
}
```
Post if you get any errors during the install, there were some new quirks with the new releases (reminds me to look at https://github.com/OpenMage/OpenMage.github.io/pull/118 )

**Reply by johnvoncolln** [reply]
Thanks, I'll take a look at that article later today or tonight. But this is for a legacy store, probably not newer projects

**Reply by justinbeaty** [reply]
@johnvoncolln If you're not intending to put a lot of effort into the legacy store, you probably can use the original Gene version where you can just copy and paste the module into the source tree:

ZIP: https://github.com/justinbeaty/module-gene-braintree/archive/e0793fb2861abde2084d285cd8bb7653ad3dd640.zip

Source: https://github.com/justinbeaty/module-gene-braintree/tree/e0793fb2861abde2084d285cd8bb7653ad3dd640

It uses an older Braintree SDK that is technically unsupported, but I don't think there's any API differences.

**Comment by johnvoncolln** [comment]
Oh I got yours working with composer and am making calls to the Braintree sandbox, just wondering about the openmage codebase as a whole and updating in the future... I might just commit the vendor folder to avoid any headaches when pushing to prod though.

**Reply by justinbeaty** [reply]
That's actually exactly what I do, because I don't want to deal with composer on the server. I must have misunderstood what you were asking about. 

Here's what I do exactly:

- I have a git repo called `magento-root` that has exactly four files:
  - composer.json, composer.lock
  - package.json, yarn.lock
- This repo also has a submodule, `magento-dist` at `htdocs`

Don't ask exactly why I brought nodejs into this, it's just because I have all other projects with a similar build / deploy command. You could probably do the same with composer scripts (or just a simple build.sh file)

Here's my package.json:

```
{
  "name": "magento-root",
  "version": "1.8.2",
  "description": "",
  "scripts": {
    "composer": "composer install --no-dev",
    "update": "composer update --no-dev",
    "build": "$(mv htdocs/app/etc/local.xml htdocs/local.xml || true) && find htdocs -mindepth 1 -maxdepth 1 -not -name .git -not -name .gitignore -not -name index.php -not -name local.xml -not -name media -not -name var -exec rm -rf {} \\; && rm -rf vendor && composer install --no-dev && $(mv htdocs/local.xml htdocs/app/etc/local.xml || true) && cp -R vendor htdocs && find htdocs -mindepth 2 -name .git -prune -exec rm -rf {} \\;",
    "deploy": "if [ -n \"$(git status --porcelain=v1 2>/dev/null)\" ]; then echo \"You have unstaged commits, exiting\"; else msg=\"Deploy: Version $(cat package.json | json version) on $(date)\" && cd htdocs && git add -A && git commit -m \"$msg\" && git push && cd - && git add htdocs && git commit -m \"$msg\" && git push; fi"
  },
  "dependencies": {},
  "devDependencies": {
    "json": "^10.0.0"
  }
}
```

Here's how I use it:

```
# initial composer install, don't ever need to run this except for after clone
yarn composer

# Update magento-lts, and all modules / deps
yarn update

# Delete all files (except /index.php, /media/, /var/, /app/etc/local.xml) and rebuild 
yarn build

# Note: Once you've rebuilt and tested locally, git add composer.json / composer.lock, bump version in package.json, and commit like "update modules & version bump"

# Push the submodule to upstream
yarn deploy
```


Then on the server `git pull` and if necessary `n98-magerun cache:flush` and that's it. :)

Hopefully that helps, I've been meaning to write that down somewhere because I think it's useful.

**Reply by johnvoncolln** [reply]
Yeah that's helpful for sure - do you know if the Kount integration on the Braintree module works?

**Reply by justinbeaty** [reply]
@johnvoncolln I am not sure as I've never used it. I presume it did work originally when Gene built the module, but I don't know exactly how it integrates with Braintree.
